manifest.go

```diff
@@ -1477,131 +1470,68 @@ func (d *dataFile) EqualityFieldIDs() []int {
-func (m *manifestEntryV1) SnapshotID() int64           { return m.Snapshot }
-
-func (m *manifestEntryV1) SequenceNum() int64 {
-	if m.SeqNum == nil {
-		return 0
 	}
-
-	return *m.SeqNum
 }
 
-func (m *manifestEntryV1) FileSequenceNum() *int64 {
-	return m.FileSeqNum
-}
-
-func (m *manifestEntryV1) DataFile() DataFile { return m.Data }
-
-func (m *manifestEntryV1) wrap(status ManifestEntryStatus, snapshotID int64, seqNum int64, fileSeqNum *int64, datafile DataFile) ManifestEntry {
-	m.EntryStatus = status
-	m.Snapshot = snapshotID
-	if seqNum > 0 {
-		m.SeqNum = &seqNum
-	}
-
-	m.FileSeqNum = fileSeqNum
-
-	m.Data = datafile
-
-	return m
-}
-
-// ManifestEntryV2Builder is a helper for building a V2 manifest entry
-// struct which will conform to the ManifestEntry interface.
-type ManifestEntryV2Builder struct {
-	m *manifestEntryV2
-}
-
-// NewManifestEntryV2Builder is passed all of the required fields and then allows
-// all of the optional fields to be set by calling the corresponding methods
-// before calling [ManifestEntryV2Builder.Build] to construct the object.
-func NewManifestEntryV2Builder(status ManifestEntryStatus, snapshotID int64, data DataFile) *ManifestEntryV2Builder {
-	return &ManifestEntryV2Builder{
-		m: &manifestEntryV2{
-			EntryStatus: status,
-			Snapshot:    &snapshotID,
-			Data:        data,
-		},
-	}
-}
-
-// SequenceNum sets the sequence number for the manifest entry.
-func (b *ManifestEntryV2Builder) SequenceNum(num int64) *ManifestEntryV2Builder {
+func (b *ManifestEntryBuilder) SequenceNum(num int64) *ManifestEntryBuilder {
 	b.m.SeqNum = &num
 
 	return b
 }
 
-// FileSequenceNum sets the file sequence number for the manifest entry.
-func (b *ManifestEntryV2Builder) FileSequenceNum(num int64) *ManifestEntryV2Builder {
+func (b *ManifestEntryBuilder) FileSequenceNum(num int64) *ManifestEntryBuilder {
 	b.m.FileSeqNum = &num
 
 	return b
 }
 
-// Build returns the constructed manifest entry, after calling Build this
-// builder should not be used further as we avoid copying by just returning
-// a pointer to the constructed manifest entry. Further calls to the modifier
-// methods after calling build would modify the constructed ManifestEntry.
-func (b *ManifestEntryV2Builder) Build() ManifestEntry {
+func (b *ManifestEntryBuilder) Build() ManifestEntry {
 	return b.m
 }
 
-type manifestEntryV2 struct {
+type manifestEntry struct {
 	EntryStatus ManifestEntryStatus `avro:"status"`
 	Snapshot    *int64              `avro:"snapshot_id"`
 	SeqNum      *int64              `avro:"sequence_number"`
 	FileSeqNum  *int64              `avro:"file_sequence_number"`
 	Data        DataFile            `avro:"data_file"`
 }
 
-func (m *manifestEntryV2) inheritSeqNum(manifest ManifestFile) {
+func (m *manifestEntry) Status() ManifestEntryStatus { return m.EntryStatus }
+func (m *manifestEntry) SnapshotID() int64 {
+	if m.Snapshot == nil {
+		return -1
```

Comment:
just a callout, i think its fine to write a `-1` here, but it might be better to write `null` /`nil`
https://github.com/apache/iceberg/pull/12334/files#diff-36347a47c3bf67ea2ef6309ea96201814032d21bb5f162dfae4045508c15588aR1763

Reply:
it will write `nil` when it writes the manifest entry. this is just for avoiding potential nil pointer dereferences for consumers that retrieve the SnapshotID from the entry.

It also makes things easier as consumers don't need to do something like:

```go
if entry.SnapshotID() != nil && *entry.SnapshotID() > 0 {

}
```

or otherwise jump through hoops to check for nil. They can just use <= 0 vs > 0 to determine if it's a valid snapshot id or if one was missing.

Reply:
ah thanks! make sense